### PR TITLE
feat: radio component

### PR DESCRIPTION
**Pull request by BogdanDanilescu**
### Implementation of the Radio component ( React & HTML )

PR for adding the Radio component for the HTML and React library.

- add global CSS classes
- add stories
- add tests
- add example in Wagtail, vite, nextjs
- add documentation

props:
```
fieldId
inline
items
errorMessage
dividerOption
title
size
```

_**Stories**_
image
Screenshot 2024-10-10 at 11 45 30
Screenshot 2024-10-10 at 11 45 39
Screenshot 2024-10-10 at 11 45 49
Screenshot 2024-10-10 at 11 45 59
Screenshot 2024-10-10 at 11 46 07
Screenshot 2024-10-10 at 11 46 16

### CODE REVIEW

File: apps/docs/content/3-components/2-library/31-radio-button.mdx [hunk 1]
```diff
@@ -2,13 +2,113 @@
...
+<Tabs id="tab1">
+  <TabList>
+    <TabItem value="tab11" checked>HTML</TabItem>
+    <TabItem value="tab12">Macro</TabItem>
+    <TabItem value="tab13">React</TabItem>
+  </TabList>
+  <TabPanel value="tab11">
+  ```html
+<div class="govie-form-group "></div>
+  <legend class="govie-fieldset__legend govie-fieldset__legend--l">
```
Comment (pregno) [inline apps/docs/content/3-components/2-library/31-radio-button.mdx:27]: What are those `govie_*` classes ?

Comment (BogdanDanilescu) [reply apps/docs/content/3-components/2-library/31-radio-button.mdx:27]: That is the example for the local HTML library, not global one. The classes should be declared in the old repo
https://github.com/ogcio/ogcio-ds

Comment (pregno) [reply apps/docs/content/3-components/2-library/31-radio-button.mdx:27]: We have to use our global classes for that.
Check how we did for the heading or paragraph.

Comment (BogdanDanilescu) [reply apps/docs/content/3-components/2-library/31-radio-button.mdx]: fixed

**Review by pregno — APPROVED** [review]

File: packages/html/ds/src/radio/radio.schema.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        .optional(),
+      value: zod.string({
+        description:
+          'The value associated with the input (this is also the value that is sent on submit)',
+        required_error: 'The value associated with the input is required',
+      }),
+      hint: zod
+        .string({
+          description:
+            'if there is additional text required in order to give the user more context',
```
Comment (hamza14khan) [inline packages/html/ds/src/radio/radio.schema.ts:37]: minor: Should start with capital letter `If ...`

File: packages/html/ds/src/radio/radio.schema.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    .array(),
+  errorMessage: zod
+    .string({
+      description: 'The error message to be displayed',
+    })
+    .optional(),
+  dividerOption: zod
+    .object({
+      label: zod.string({
+        description: 'the value of the divider radio label',
```
Comment (hamza14khan) [inline packages/html/ds/src/radio/radio.schema.ts:56]: Same as above for consistency.

File: packages/html/ds/src/radio/radio.stories.ts [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+
+export default meta;
+type Story = StoryObj<typeof meta>;
+
+export const Default: Story = {
+  argTypes: {
+    fieldId: {
+      control: 'text',
+      type: { name: 'string', required: true },
+      description: 'the unique value for the radios group',
```
Comment (hamza14khan) [inline packages/html/ds/src/radio/radio.stories.ts:56]: Same as above.

File: packages/react/ds/src/heading/heading.tsx [hunk 5]
```diff
@@ -6,13 +6,13 @@ export function Heading({
   size,
   children,
   caption,
-  className,
+  customClasses = '',
```
Comment (hamza14khan) [inline packages/react/ds/src/heading/heading.tsx:9]: You might wanna resolve conflicts here 😄

**Review by hamza14khan — APPROVED** [review]
Minor comments but overall looks good.

**Review by ramonmulia — APPROVED** [review]

**Pull request merged**